packages/hub-threads-client/src/client.ts

```diff
@@ -66,6 +64,33 @@ Client.prototype.getThread = async function (
   })
 }
 
+// Private listDBs method
+const oldListDBs = Client.prototype.listDBs
+
+/**
+ * Lists all known DBs.
+ */
+// eslint-disable-next-line @typescript-eslint/ban-ts-ignore
+// @ts-ignore
+Client.prototype.listDBs = async function (
+  ctx?: Context,
+): Promise<Record<string, GetThreadResponse | undefined>> {
+  const dbs: Record<string, GetThreadResponse | undefined> = {}
+  if (ctx?.get('x-textile-api-sig')) {
```

Comment:
the _only_ thing i can think... if they call this before the sig has been created will they get a helpful error or will it say the api doesn't exist?

Reply:
Cool! This made me realize... since the dashboard code will in some way need to list threads, etc., it probably makes sense to utilize this code. We'll just have to add handling for normal dev/org sessions (like the CLI), which can be stored in a browser cookie / local storage.

Reply:
no actually I'm wrong, they'll get the right error message... that they haven't authenticated or whatever

Reply:
oh, i thought it would see it missing so try to call the other api. which wouldn't exist

Reply:
yup, that's right... I'm crazy. added "catch" for this situation.